weave/integrations/google_ai_studio/google_ai_studio_sdk.py

```diff
@@ -32,7 +32,10 @@ def gemini_accumulator(
 def gemini_on_finish(
     call: Call, output: Any, exception: Optional[BaseException]
 ) -> None:
-    original_model_name = call.inputs["self"].model_name
+    if hasattr(call.inputs["self"], "model_name"):
+        original_model_name = call.inputs["self"].model_name
+    elif hasattr(call.inputs["self"], "model"):
+        original_model_name = call.inputs["self"].model.model_name
```

Comment:
what about the case where it has neither?  Is that valid (then you need a default), or invalid (then you need an error msg)

Reply:
Added error message.